torch/amp/__init__.py

```diff
@@ -1,3 +1,13 @@
+__all__ = [
+    "GradScaler",
+    "_enter_autocast",
+    "_exit_autocast",
+    "autocast",
+    "custom_bwd",
+    "custom_fwd",
+    "is_autocast_available",
+]
+
 from .autocast_mode import (
```

Comment:
nit: `__all__` should place after the import statements rather than at the top of the file.

Reply:
Is that a pytorch specific guideline? Because according to PEP8 it should come before imports. (other than `__future__`).
https://peps.python.org/pep-0008/#module-level-dunder-names

Reply:
References:

https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/wrong-import-position.html

> wrong-import-position / C0413
> 
> Message emitted:
> 
> Import "%s" should be placed at the top of the module

https://pycqa.github.io/isort/docs/configuration/options.html#float-to-top

> Float To Top
> 
> Causes all non-indented imports to float to the top of the file having its imports sorted (immediately below the top of file comment). This can be an excellent shortcut for collecting imports every once in a while when you place them in the middle of a file to avoid context switching.

Reply:
Putting it before imports makes more sense to me because I think of `__all__` like a header for the module. It tells me what the module is going to define/export and everything after that is just the implementation details.

Reply:
If you run `pylint` on the example from the PEP8 spec, no cases of C0413 are raised. These are intentionally excluded as per PEP8 spec: https://github.com/pylint-dev/pylint/blob/6bb335f5ca2c52a994d5836eee63dc2c140fc5f3/pylint/checkers/imports.py#L637-L645